src/main/java/com/microsoft/aad/msal4j/ADFSAuthority.java

```diff
@@ -9,14 +9,23 @@
 
     final static String AUTHORIZATION_ENDPOINT = "oauth2/authorize";
     final static String TOKEN_ENDPOINT = "oauth2/token";
+    final static String DEVICE_CODE_ENDPOINT = "oauth2/devicecode";
 
     private final static String ADFS_AUTHORITY_FORMAT = "https://%s/%s/";
+    private final static String DEVICE_CODE_ENDPOINT_FORMAT = ADFS_AUTHORITY_FORMAT + DEVICE_CODE_ENDPOINT;
+
+    String deviceCodeEndpoint;
```

Comment:
I think you should be able to move need this and the getter to 'Authority'. You can also get rid of 'deviceCodeEnpoint' in AADAuthority as well, and just set it on the parent class based on the child classes format.

Reply:
Fixed this and your other comments in the latest commit.